source/elements/oneMKL/source/domains/spblas/operations/spsv.rst

```diff
@@ -283,8 +285,8 @@ spsv
 
    A_view
       Specifies which part of the handle should be read as described by
-      :ref:`onemkl_sparse_matrix_view`. ``A_view.type_view`` must be
-      ``matrix_descr::triangular`` or ``matrix_descr::diagonal``.
+      :ref:`onemkl_sparse_matrix_view`. The ``type_view`` field must be
+      ``matrix_descr::triangular``.
```

Comment:
TRSV also does take in and support the `diagonal` field, so please add that in. Both oneMKL (see [oneapi::mkl::sparse::trsv()](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2024-1/oneapi-mkl-sparse-trsv.html)) and cuSPARSE (see [cusparseSpMatSetAttribute()](https://docs.nvidia.com/cuda/cusparse/#cusparsespmatsetattribute) that must be called before cusparseSpSV()) support that.

Reply:
Also add in `Support may be expanded in the future`

Reply:
The `diag_view` field is supported when the `matrix_view` is `triangular`. The `diag_view` field is tested and works.
This change means that there is no way for the user to indicate to any of the backends that the input matrix only has values in the `diagonal`.
I don't like adding that support may be expanded in the future, this is true for pretty much all the limitations in the spec. I think this is just a detail that hinder readability.

Reply:
Ignore the support to be added in the future; that's fine.

> This change means that there is no way for the user to indicate to any of the backends that the input matrix only has values in the diagonal.

What you mentioned would be a user-guaranteed `matrix_property`. Since we are talking about `matrix_view`s here, I believe you meant, "there is no way for user to indicate to any of the backends to read and use only the diagonal values (among others) in the input sparse matrix". That would be correct, because no backends support `matrix_descr type_view = matrix_descr::diagonal` in the `matrix_view` structure right now for any spec APIs.

You mentioned in `matrix_view.rst`:
>    Each operation documents which combination of ``type_view``, ``uplo_view`` and ``diag_view`` are valid.

That's why I am asking for us to add that `matrix_descr type_view = matrix_descr::triangular` and `diag diag_view = diag::{unit, nonunit}` must **both** be specified with `spsv`. What we have here in this PR is only about the ``type_view`` field; info about `diag diag_view=?` field is missing here.

Reply:
Maybe there's some confusion: `diag diag_view` is not specific to `matrix_descr::diagonal` alone. `diag diag_view` is also applicable to and frequently used with `matrix_descr::triangular`.

Reply:
What we need here is:
```
The ``type_view`` field must be ``matrix_descr::triangular``. The ``diag_view`` must be specified to ``diag::nonunit`` or ``diag::unit``
```
On a related note, at the bottom of the [Matrix view page](https://uxlfoundation.github.io/oneAPI-spec/spec/elements/oneMKL/source/domains/spblas/matrix_view.html#onemkl-sparse-matrix-descriptor), we have:
```
If the matrix_desc is diagonal, diag_view is initialized to diag::unit.
```
The default for `diag_view` should always be `nonunit`. We want users to intentionally choose `unit` because that is a rarer usage as opposed to `nonunit`. Do you mind fixing that in this PR?

Reply:
My intention was to only document when a field is ignored or when it cannot be set to a value but it could maybe cause some confusion. I've added a bit of clarification in https://github.com/uxlfoundation/oneAPI-spec/pull/542/commits/96b30890d61c52b31a88bb616faa80f1e198193c.
It's not quite your suggestion of "diag_view must be specified" since the field is already defaulted to `nonunit` see [here](https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneMKL/source/domains/spblas/matrix_view.rst?plain=1#L68).